### PR TITLE
Mill: remove vases, remove `mem` from `cart`, rename `zygote` to `embryo`, recompile smart-lib

### PR DESCRIPTION
**Pull request by dr-frmr**

### CODE REVIEW

File: app/sequencer.hoon [hunk 1]
```diff
@@ -270,10 +270,10 @@
     ``noun+!>(%.y)
   ::
       [%rice @ ~]
-    (read-rice path 0 (need town-id.state) p.town.state)
+    (read-rice +.path 0 (need town-id.state) p.town.state)
```
Comment (tacryt-socryp) [inline app/sequencer.hoon:273]: Given that `path` is a list, we can refer to `i` for list items and `t` for the rest of the list. I'd prefer use of the faces to lark syntax for list accesses (and most of the time).

**Review by tacryt-socryp — APPROVED** [review]

**Comment by hosted-fornet** [comment]
I can now successfully start a town using

```
:sequencer|init 1
```

However, I can't send zigs on town 1 using the line suggested in the README:

```
:wallet &zig-wallet-poke [%submit 0x3.e87b.0cbb.431d.0e8a.2ee2.ac42.d9da.cab8.063d.6bb6.2ff9.b2aa.e1b9.0f56.9c3f.3423 0x74.6361.7274.6e6f.632d.7367.697a 1 [1 10.000] [%give 1.936.157.050 0x2.eaea.cffd.2bbe.e0c0.02dd.b5f8.dd04.e63f.297f.14cf.d809.b616.2137.126c.da9e.8d3d 777]]
```

which gives

```
>   [ caller=[id=0x3.e87b.0cbb.431d.0e8a.2ee2.ac42.d9da.cab8.063d.6bb6.2ff9.b2aa.e1b9.0f56.9c3f.3423 nonce=2 zigs=0xcb22.90ff.6e2f.ba63.86e1.40bb.6e89.c2d8]                                     
    args                                                                                                                                                                                         
  [~ [1.702.259.047 337.840.262.743.242.111.263.054.621.221.169.010.230.914.627.789.838.976.422.253.503.447.446.186.790.205 [0 270.012.762.141.793.524.098.028.425.168.146.792.833] 777 10.000]] 
    grains
  { [ p=0xcb22.90ff.6e2f.ba63.86e1.40bb.6e89.c2d8
        q
      [ id=0xcb22.90ff.6e2f.ba63.86e1.40bb.6e89.c2d8
        lord=0x74.6361.7274.6e6f.632d.7367.697a
        holder=0x3.e87b.0cbb.431d.0e8a.2ee2.ac42.d9da.cab8.063d.6bb6.2ff9.b2aa.e1b9.0f56.9c3f.3423
        town-id=1
        germ=[%.y p=[salt=1.936.157.050 data=[10.321.055.000.000.000.000 0 7.721.149.764.861.789.085.613.156.690.298]]]
      ]
    ]
  }
]
>>> [ me=0x74.6361.7274.6e6f.632d.7367.697a
  block=8
  town-id=1
    owns
  { [ p=0xcb22.90ff.6e2f.ba63.0d88.3e91.f6da.0181
        q
      [ id=0xcb22.90ff.6e2f.ba63.0d88.3e91.f6da.0181
        lord=0x74.6361.7274.6e6f.632d.7367.697a
        holder=0x2.eaea.cffd.2bbe.e0c0.02dd.b5f8.dd04.e63f.297f.14cf.d809.b616.2137.126c.da9e.8d3d
        town-id=1
        germ=[%.y p=[salt=1.936.157.050 data=[50.000.000.000.000.000.000 0 7.721.149.764.861.789.085.613.156.690.298]]]
      ]
    ]
  }
]
>>  "write result: [[%.n p=~[[%rose p=[p=\":\" q=\"\" r=\"\"] q=[i=[%rose p=[p=\"/\" q=\"/\" r=\"\"] q=[i=[%leaf p=\"lib\"] t=[i=[%leaf p=\"zig\"] t=[i=[%leaf p=\"mill\"] t=[i=[%leaf p=\"hoon\"] t=~]]]]] t=[i=[%leaf p=\"<[211 29].[211 60]>\"] t=~]]] [%rose p=[p=\":\" q=\"\" r=\"\"] q=[i=[%rose p=[p=\"/\" q=\"/\" r=\"\"] q=[i=[%leaf p=\"lib\"] t=[i=[%leaf p=\"zig\"] t=[i=[%leaf p=\"mill\"] t=[i=[%leaf p=\"hoon\"] t=~]]]]] t=[i=[%leaf p=\"<[211 20].[211 61]>\"] t=~]]]]] 9.993]"
took 267741µs
```

**Comment by dr-frmr** [comment]
> I can now successfully start a town using
> 
> ```
> :sequencer|init 1
> ```
> 
> However, I can't send zigs on town 1 using the line suggested in the README:
> 

Had to re-recompile all the contracts so they'd work again, committing updated versions now

**Comment by hosted-fornet** [comment]
Other than starting a new town, basic txs are still not working for me: all are rejected by `+mule`. I tried recompiling a contract and couldn't because `json` type was removed from our `hoon.hoon`, I think in [this commit](https://github.com/uqbar-dao/ziggurat/commit/d2bfec6e0574e8a5e72325cd753c08aa93b27223) (see, e.g., [here](https://github.com/uqbar-dao/ziggurat/blob/d2bfec6e0574e8a5e72325cd753c08aa93b27223/lib/zig/sys/hoon.hoon)).

Is the plan to add back `json` typing later or in a different place? Are the tx.s working for you? Maybe I'm just doing something wrong here?

**Comment by dr-frmr** [comment]
> Other than starting a new town, basic txs are still not working for me: all are rejected by `+mule`. I tried recompiling a contract and couldn't because `json` type was removed from our `hoon.hoon`, I think in [this commit](https://github.com/uqbar-dao/ziggurat/commit/d2bfec6e0574e8a5e72325cd753c08aa93b27223) (see, e.g., [here](https://github.com/uqbar-dao/ziggurat/blob/d2bfec6e0574e8a5e72325cd753c08aa93b27223/lib/zig/sys/hoon.hoon)).
> 
> Is the plan to add back `json` typing later or in a different place? Are the tx.s working for you? Maybe I'm just doing something wrong here?

Huh, I see what you mean about `json` being removed but I was somehow able to compile these contracts and run them in this branch as of that latest commit

**Comment by hosted-fornet** [comment]
Okay, weird. If it's working for you, let's merge.

**Review by hosted-fornet — APPROVED** [review]

File: lib/zig/mill.hoon [hunk 2]
```diff
@@ -199,10 +199,12 @@
       ++  weed
         |=  [to=id budget=@ud]
         ^-  [(unit chick) rem=@ud =errorcode]
-        =/  cart  [to blocknum town-id owns.crop]
+        =/  =cart  [to blocknum town-id owns.crop]
         ::  TODO figure out how to pre-cue this and get good results
         ::
         =/  =contract  (hole contract [nok.crop +:(cue q.q.smart-lib)])
+        ~&  >  embryo
+        ~&  >>>  cart
```
[the remark this answers is not in the document]
Comment (dr-frmr) [reply lib/zig/mill.hoon:207]: No, I'll remove. But I also wanna work out this JSON thing before we merge. I think our hoon.hoon should have it

[the remark this answers is not in the document]
Comment (tacryt-socryp) [reply lib/zig/mill.hoon:207]: Can you put the JSON stuff in a different file? Smart would be fine (if it's in its own core). I'd really like our hoon.hoon to stay virginal/minimally modified for as long as possible.

[the remark this answers is not in the document]
Comment (dr-frmr) [reply lib/zig/mill.hoon:207]: Ok yeah, no problem. I'll add the `json` type to smart along with the `enjs` part of `format`

**Comment by dr-frmr** [comment]
Added JSON stuff to smart.hoon, made a new smart-lib.noun, and recompiled all contracts against it (left DAO and multisig contracts alone since I know those are being worked on in other branches). 

I am noticing now that wallet might not be getting updates for all towns from indexer. I probably have to re-align it with the latest update format, will do that in a new wallet-focused PR tomorrow. I think this one is good to go.

If it doesn't work for whatever reason, try starting up a new fakeship, because they like to cache the /* imports it seems.

**Pull request merged**